source/beamline/REIXS/REIXSBeamline.cpp

```diff
@@ -874,8 +874,8 @@ AMControl::FailureExplanation REIXSBrokenMonoControl::move(double setpoint)
 bool REIXSBrokenMonoControl::stop()
 {
 	if(moveAction_) {
-		moveAction_->cancel();
 		stopInProgress_ = true;
+		moveAction_->cancel();
```

Comment:
This is the key change. We need to set the `stopInProgress_` flag before call the `moveAction_->cancel()`